File: interpreter/java/AccountingExpression.java
package Interpreter;

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
class AccountingExpression implements IJobExpression {
    public void interpret(Context context) {
        if (context.formula.contains("M")) {
            context.totalPoint += 3000;
        }
    }
}

File: interpreter/kotlin/AccountingExpression.kt
package Interpreter

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
internal class AccountingExpression : IJobExpression {
    override fun interpret(context: Context) {
        if (context.formula!!.contains("M")) {
            context.totalPoint = context.totalPoint + 3000
        }
    }
}